packages/api/src/external/fhir/patient/count-resources.ts

```diff
@@ -0,0 +1,89 @@
+import { Patient } from "../../../models/medical/patient";
+import { capture } from "../../../shared/notifications";
+import { Util } from "../../../shared/util";
+import { makeFhirApi } from "../api/api-factory";
+import { ResourceTypeForConsolidation } from "./consolidated";
+import { fullDateQueryForResource, getPatientFilter } from "./resource-filter";
+
+export type ResourceCount = {
+  total: number;
+  resources: {
+    [key in ResourceTypeForConsolidation]?: number;
+  };
+};
+
+export async function countResourcesPerPatient({
+  patient,
+  resources = [],
+  dateFrom,
+  dateTo,
+}: {
+  patient: Pick<Patient, "id" | "cxId">;
+  resources?: ResourceTypeForConsolidation[];
+  dateFrom?: string;
+  dateTo?: string;
+}): Promise<ResourceCount> {
+  const { log } = Util.out(
+    `countResourcesPerPatient - cxId ${patient.cxId}, patientId ${patient.id}`
+  );
+  const fhir = makeFhirApi(patient.cxId);
+
+  const {
+    resourcesByPatient,
+    resourcesBySubject,
+    dateFilter: fullDateQuery,
+  } = getPatientFilter({
+    resources,
+    dateFrom,
+    dateTo,
+  });
+
+  const summaryCount = "&_summary=count";
+
+  const result = await Promise.allSettled([
+    ...resourcesByPatient.map(async resource => {
+      const dateFilter = fullDateQueryForResource(fullDateQuery, resource);
+      const filter = `patient=${patient.id}${dateFilter}${summaryCount}`;
+      const res = await fhir.search(resource, filter);
+      return { resourceType: resource, count: res.total ?? 0 };
+    }),
+    ...resourcesBySubject.map(async resource => {
+      const dateFilter = fullDateQueryForResource(fullDateQuery, resource);
+      const filter = `subject=${patient.id}${dateFilter}${summaryCount}`;
+      const res = await fhir.search(resource, filter);
+      return { resourceType: resource, count: res.total ?? 0 };
+    }),
+  ]);
```

Comment:
Could be a good opportunity to use the [$everything](https://smilecdr.com/docs/fhir_repository/reading_data.html#everything-operation) operation here

Reply:
Yeah, I tried this in `prod` for the cx/patient w/ large volume of data, but the result doesn't have the total:

<img width="273" alt="image" src="https://github.com/metriport/metriport/assets/2132564/3d0dc3e9-f73a-4680-863e-3419df275792">

I'm assuming its an error/issue w/ HAPi in some situations, since this works local:

<img width="298" alt="image" src="https://github.com/metriport/metriport/assets/2132564/db809816-7d6e-4d50-bd2c-fe66fea3a4f8">